include/mbedtls/bignum.h

```diff
@@ -740,21 +741,32 @@ int mbedtls_mpi_is_prime( const mbedtls_mpi *X,
                   int (*f_rng)(void *, unsigned char *, size_t),
                   void *p_rng );
 
+/**
+ * \brief Flags for mbedtls_mpi_gen_prime()
+ *
+ * Each of these flags is a constraint on the result X returned by
+ * mbedtls_mpi_gen_prime().
+ */
+typedef enum {
+    MBEDTLS_MPI_GEN_PRIME_FLAG_DH = 0x0001,     /**< (X-1)/2 is prime too */
+    MBEDTLS_MPI_GEN_PRIME_CHECK_RNG = 0x0004   /**< check RNG validity (by number of iterations required for prime generation) Note: 1% false positive probability*/
```

Comment:
I am sorry I made a mistake in my calculations of the false positive probability. It is much less than 1% the correct estimate is around 0.07%.

Reply:
0.07% is very high. To give just one real-life consequence: at 6 RSA key generations per test suite run and 26 test suite runs per CI job, that's about 10% of the CI jobs failing.

Fortunately I think your calculations are off. If we filter for range [n/sqrt(2), n] and we filter out even integers, and only reject integers based on primality, then the probability of hitting a prime is p(n) = (π(n) - π(n/√2)) / (n - n/√2) ≈ (n / ln(n) - n/√2/(ln(n) - ln(√2))) / (n - n/√2). If we make 5*N attempts for n=2^N then the probability of consistently failing is r(N) given by
```
p = lambda n: 2 * (1/log(n) - 1/sqrt(2)/(log(n)-log(sqrt(2)))) / (1-1/sqrt(2))
r = lambda N: (1 - p(2**N)) ** (5*N)
```
r(1024) ≈ 5.4e-7, i.e. 2^{-21}. It's small enough that we're unlikely to see it failing randomly during our test runs, but if you deploy it to a trillion devices and they each generate a key, that's still a million failures. 2^{-21} is not negligible by cryptographic standards.